File: 2017/12/spec/runner.spec.js
import { parse } from "../lib/parser";
import { part1, part2 } from "../lib/runner";

const SAMPLE = ` 0 <-> 2
    1 <-> 1
    2 <-> 0, 3, 4
    3 <-> 2, 4
    4 <-> 2, 3, 6
    5 <-> 6
    6 <-> 4, 5`

describe("Runner", () => {
  test("part 1 passes tests cases", () => {
    expect(part1(parse(SAMPLE))).toEqual(6)
  });

  test("part 2 passes tests cases", () => {
    expect(part2(parse(SAMPLE))).toEqual(2)
  });
});

File: 2020/3/spec/runner_spec.rb
# frozen_string_literal: true

require "runner"
require "parser"

describe Runner do
  it "calculates the correct answer for the test input for part1" do
    expect(Runner.part_1(Parser.parse(sample_input))).to eq(7)
  end

  it "calculates the correct answer for the test input part2" do
    expect(Runner.part_2(Parser.parse(sample_input))).to eq(336)
  end

  def sample_input
    <<~INPUT.strip
    ..##.......
    #...#...#..
    .#....#..#.
    ..#.#...#.#
    .#...##..#.
    ..#.##.....
    .#.#.#....#
    .#........#
    #.##...#...
    #...##....#
    .#..#...#.#
    INPUT
  end
end
